android/src/main/java/com/pagopa/ioreactnativecrypto/X509VerificationUtil.kt

```diff
@@ -0,0 +1,300 @@
+          return CertificateValidationResult(
+            isValid = false,
+            validationStatus = CertificateValidationStatus.EXPIRED,
+            errorMessage = "Certificate expired: ${cert.subjectX500Principal}"
+          )
+        } catch (e: CertificateNotYetValidException) {
+          return CertificateValidationResult(
+            isValid = false,
+            validationStatus = CertificateValidationStatus.NOT_YET_VALID,
+            errorMessage = "Certificate not yet valid: ${cert.subjectX500Principal}"
+          )
+        }
+      }
+
+      // Collect all CRLs from distribution points in the certificate chain
+      val crls = mutableListOf<X509CRL>()
+      for (cert in certificateChain) {
+        crls.add(downloadCertificateCrl(certificateFactory, cert))
+      }
+
+      // Check each certificate against the CRLs
+      for (cert in certificateChain) {
+        for (crl in crls) {
+          // Only check against CRLs issued by the certificate's issuer
+          if (cert.issuerX500Principal == crl.issuerX500Principal) {
+            val revokedCert = crl.getRevokedCertificate(cert)
+            if (revokedCert != null) {
+              return CertificateValidationResult(
+                isValid = false,
+                validationStatus = CertificateValidationStatus.REVOKED,
+                errorMessage = "Certificate revoked: ${cert.subjectX500Principal}, revocation date: ${revokedCert.revocationDate}"
+              )
+            }
+          }
+        }
+      }
+
+      // Create a CertPath from the certificate chain.
+      val certPath = certificateFactory.generateCertPath(certificateChain)
+
+      // Set up PKIX parameters using the trust anchor.
+      val pkixParams = PKIXParameters(setOf(trustAnchor)).also {
+        it.isRevocationEnabled = crls.isNotEmpty()
+      }
+
+      // Add CRLs to CertStore if any were downloaded
+      if (crls.isNotEmpty()) {
+        val certStore = CertStore.getInstance(
+          "Collection",
+          CollectionCertStoreParameters(crls)
+        )
+        pkixParams.addCertStore(certStore)
+      }
+
+      // Validate the certificate chain.
+      val validator = CertPathValidator.getInstance("PKIX")
+      validator.validate(certPath, pkixParams)
+
+      return CertificateValidationResult(
+        isValid = true,
+        validationStatus = CertificateValidationStatus.VALID
+      )
+    } catch (e: CertPathValidatorException) {
+      return CertificateValidationResult(
+        isValid = false,
+        validationStatus = CertificateValidationStatus.INVALID_CHAIN,
+        errorMessage = "Certificate path validation failed: ${e.message}"
+      )
+    } catch (e: VerificationException) {
+      return when (e) {
+        VerificationException.CRLExpired ->
+          CertificateValidationResult(
+            isValid = false,
+            validationStatus = CertificateValidationStatus.CRL_EXPIRED,
+            errorMessage = "CRL not defined"
```

Comment:
```suggestion
            validationStatus = CertificateValidationStatus.CRL_EXPIRED,
            errorMessage = "CRL expired"
```

Reply:
BTW, what about adding in the error message the raw value of the enum or use a function that give you the message according to the enum value?  What do you think?